modules/distribution/product/src/main/startup-scripts/profileSetup.sh

```diff
@@ -0,0 +1,296 @@
+	fi
+}
+
+disableIndexingConfiguration(){
+	value=`xmllint --xpath 'wso2registry/indexingConfiguration/startIndexing/text()' $pathToRegistry`
+	if [ "$value" = "true" ]
+	then
+		sed -i "/<indexingConfiguration>/,/<\/indexingConfiguration>/ s/<startIndexing>true<\/startIndexing>/<startIndexing>false<\/startIndexing>/g;" $pathToRegistry
+		timeStamp
+  	echo "[${timestamp}] INFO - Disable the indexingConfiguration from registry.xml"
+	elif [ "$value" = "false" ]
+	then
+		timeStamp
+  	echo "[${timestamp}] INFO - IndexingConfiguration from registry.xml is already configured"
+	fi
+}
+
+removeSynapseConfigs(){
+	if [ "$(find ${pathToSynapseConfigs} -type d -or -type f -not -name 'synapse.xml' | wc -l)" -gt 1 ]
+	then
+		if [ "$(find ${pathToSynapseConfigs} -type d | wc -l)" -gt 1 ]
+		then
+			rm -R ${pathToSynapseConfigs}/*/
+		fi
+
+		if [ -e ${pathToSynapseConfigs}/registry.xml ]
+		then
+			rm -r ${pathToSynapseConfigs}/registry.xml
+		fi
+		timeStamp
+  	echo "[${timestamp}] INFO - Removed the SynapseConfigs files"
+	else
+		timeStamp
+  	echo "[${timestamp}] INFO - SynapseConfigs are already removed"
+	fi
+}
+
+#main
+case $1 in
+	-Dprofile=api-key-manager)
+		disableDataPublisher
+		disableJMSConnectionDetails
+		disablePolicyDeployer
+		disableTransportSenderWS
+		disableTransportSenderWSS
+		removeWebSocketInboundEndpoint
+		removeSecureWebSocketInboundEndpoint
+		removeSynapseConfigs
+		#remove unnecessary webapps
+		if [ "$(find $pathToWebapps -type f -not \( -name 'client-registration#v0.12.war' -o -name 'authenticationendpoint.war' -o -name 'oauth2.war' -o -name 'throttle#data#v1.war' -o -name 'api#identity#consent-mgt#v1.0.war' \) | wc -l)" -gt 0 ]
+		then
+			find $pathToWebapps -type f -not \( -name 'client-registration#v0.12.war' -o -name 'authenticationendpoint.war' -o -name 'oauth2.war' -o -name 'throttle#data#v1.war' -o -name 'api#identity#consent-mgt#v1.0.war' \)  -delete
+			timeStamp
+	  	echo "[${timestamp}] INFO - Removed the unnecessary webapps for key-manager profile"
+		else
+			timeStamp
+	  	echo "[${timestamp}] INFO - Unnecessary webapps for key-manager profile are already removed"
+		fi
+		#remove jaggeryapps
+		if [ -d $pathToJaggeryapps ]
+		then
+			rm -r $pathToJaggeryapps
+			timeStamp
+	  	echo "[${timestamp}] INFO - Removed unnecessary jaggeryapps for key-manager profile"
+		else
+			timeStamp
+	  	echo "[${timestamp}] INFO - Unnecessary jaggeryapps for key-manager profile are already removed"
+		fi
+		;;
+	-Dprofile=api-publisher)
+		disableDataPublisher
+		disableJMSConnectionDetails
+		disableTransportSenderWS
+		disableTransportSenderWSS
+		removeWebSocketInboundEndpoint
+		removeSecureWebSocketInboundEndpoint
+		removeSynapseConfigs
+		#remove unnecessary webapps
+		if [ "$(find $pathToWebapps -type f -not -name 'api#am#publisher#v0.12.war' | wc -l)" -gt 0 ]
+		then
+			find $pathToWebapps -type f -not -name 'api#am#publisher#v0.12.war' -delete
```

Comment:
APIM 2.5.0 it should be v0.13.war. better to use regex for this. check other places as well

Reply:
Resolved